### ISSUE TITLE
[UX] Running cron via the Admin Bar menu always redirects to the "Status report" page

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3973]
**Steps To Reproduce**
Run cron via the "Run cron" menu item, under "Home"...

**Actual behavior**
You get redirected to `/admin/reports/status` and get a "Cron ran successfully." message

**Expected behavior**
You should stay on the same page.

- - - - - - - 
PR by @BWPanda - https://github.com/backdrop/backdrop/pull/2900

**Comment by findlabnet** [comment]
This is bug? Not sure. After run cron you can see if something going wrong on status report page. 
I run cron manually for this purpose only.

**Comment by indigoxela** [comment]
I agree with @findlabnet - this isn't really a bug.

Sometimes it could be better to _not_ get redirected, though. If - for instance - something's weird on a page (a view) and the admin want's to check if a cron run fixes it.

On the other hand the status report page could offer helpful information.

Personally, I don't have any preference regarding a redirect. Both is OK for me.

**Comment by klonos** [comment]
> ...for instance - something's weird on a page (a view) and the admin want's to check if a cron run fixes it.

That's actually the experience I've had twice before I filed this as a bug.

For the record, the D7 version of the admin_menu module only "redirects" to the "Status report" page if the core Overlay module is enabled (but we have removed the Overlay module from Backdrop). Even then, that is not redirecting exactly, as what happens is that the "Status report" page loads in an overlay/popup, which the user can easily close, to get back to the page they were when they run cron from:

<img width="838" alt="Screen Shot 2019-08-12 at 4 55 56 pm" src="https://user-images.githubusercontent.com/2423362/62870395-3e1aeb00-bd22-11e9-98d9-94e2550114fb.png">

...if the user runs cron (either via the admin_menu menu, or via the "No update info available" warning) while the Overlay module is disabled, they remain in the page they initiated the cron run from:

<img width="519" alt="Screen Shot 2019-08-12 at 4 49 27 pm" src="https://user-images.githubusercontent.com/2423362/62870601-95b95680-bd22-11e9-819f-316f78160e7e.png">

<img width="518" alt="Screen Shot 2019-08-12 at 4 49 43 pm" src="https://user-images.githubusercontent.com/2423362/62870609-99e57400-bd22-11e9-964e-4bf49afd7ae4.png">

So I still think this is a bug, and would expect this to work the same way that clearing caches via the Admin Bar menu. Lets wait for more feedback before we action this though.

**Comment by indigoxela** [comment]
> the D7 version of the admin_menu module only "redirects" to the "Status report" page if the core Overlay module is enabled

Are you sure? I always disable the overlay module in D7. Running cron via the admin_menu always redirects me to the status page.
Clearing caches doesn't.

In this case Backdrop behaves exactly the same as D7 with admin_menu.

**Comment by klonos** [comment]
> Are you sure?

You are right. I have tested this in simplytest.me, with admin_menu 7.x-3.0-rc6.

Here's what I've tried once the sandbox was spun up:

1. Head to `/admin/modules` -> disable the core Toolbar module -> enable the Admin Menu Toolbar Style module
2. Head to home page
3. clear cache via the Admin Menu -> no page redirect
4. run cron via the Admin Menu -> overlay/popup with the "Site status" page, and a `Cron run successfully` success message; no page redirect.
5. Close overly/popup -> still in the home page
6. Head to `admin/modules` -> disable the core Overlay module
7. Clear caches -> success message / no redirect
8. Run cron via the link in the "No update information available" warning that also shows up as a result of clearing caches -> success / **no redirect**
9. Run cron via the Admin Menu link -> success / **redirect** to status page.

**Comment by BWPanda** [comment]
I think people shouldn't be redirected after running cron. If the only reason is to see if there are issues after doing so, then that's what the [notification badge](https://github.com/backdrop/backdrop-issues/issues/639) is for.

Let's remove the redirection and make everything consistent (since clearing caches and running cron via other methods don't cause a redirect).

**Comment by BWPanda** [comment]
Here's a simple [PR](https://github.com/backdrop/backdrop/pull/2900).

**Comment by klonos** [comment]
> If the only reason is to see if there are issues after doing so, then that's what the notification badge is for.

👍 to that!

...the PR is very simple indeed, and works fine, but I would like us to consider what happens when cron fails to run (unfortunately, I don't know how to simulate that 🤔).

Also, there is still a `backdrop_goto('admin/reports/status');` in `system_run_cron()`. So in theory, if you run cron via any other way, the redirection still happens(??). I have tested running it via `/admin/config/system/cron` and there is no redirection though 🤷‍♂

**Comment by klonos** [comment]
...perhaps when cron fails, it makes sense to be redirecting to `/admin/config/system/cron`? Not sure.

**Comment by BWPanda** [comment]
So there's three places that run cron:

- Admin bar (via `system_run_cron()`)
- Cron page (`/admin/config/system/cron`) (via `system_run_cron_submit()`)
- Status page (via `system_run_cron()`)

`system_run_cron_submit()` just redirects to itself and `system_run_cron()` redirects to the Status page (without any redirection you're left on a blank menu callback page).

I've updated my [PR](https://github.com/backdrop/backdrop/pull/2900) to simplify this so that `system_run_cron_submit()` just calls `system_run_cron()`, and `system_run_cron()` now redirects back to the original page if there's a referrer, or to the Status page otherwise.

**Comment by BWPanda** [comment]
Also, if this PR doesn't get in (or my change to `system_run_cron_submit()` is reverted), then the wording in https://github.com/backdrop/backdrop/blob/1.x/core/modules/system/system.admin.inc#L1593 needs to be fixed...

**Comment by klonos** [comment]
Got bitten by this again. Here's a PR: https://github.com/backdrop/backdrop/pull/4792

It's basically 99.9% identical to @bwpanda's - just rebased and added a missing period in an inline comment to stop PHPCS from nagging.

**Comment by klonos** [comment]
@izmeez did you also mean to add the "ready to be committed" label? (meaning that you have done a code review as well?) ...this was previously set to RTBC by me, so it doesn't count - someone else needs to RTBC it.

**Comment by klonos** [comment]
OK, some slight wire-crossing there: when we remove the "needs review" label, we also add the "ready to be committed" label (if it also has the "works for me" label that is).

**Comment by klonos** [comment]
Jinx 😅

**Comment by klonos** [comment]
> I thought someone more established to Backdrop should do it.

Nope, no "established" status needed - our core committers will do the final review before merging anyway.

**Comment by klonos** [comment]
Either way is fine @izmeez 👍🏼 ...the subtle difference is this:
- `Running cron manually via the Admin Bar menu should stay on current page` sounds like a task
- `Running cron via the Admin Bar menu always redirects to the "Status report" page` sounds like a bug

**Comment by quicksketch** [comment]
I looked at this last week and let it sit to think about it. I tested it and it works as described. We do the same thing of checking HTTP_REFERER in several other places in core, like cancel links, so this is not a new pattern. I think this is a UX improvement too. I merged https://github.com/backdrop/backdrop/pull/4792 into 1.x and 1.28.x.

Thank you @klonos, @izmeez, @indigoxela, and @findlabnet for your review and feedback. Another legacy PR by @BWPanda closed out. :cry: If you see this; we still appreciate you!
